### ISSUE TITLE
[minimal] Do we want them in the "parent" profile?

### ISSUE THREAD

**Issue by udeved** [manjaro/manjaro-tools-iso-profiles#33]
We could have minimal profiles in one directory with the full profile.
For example a xfce profile could host two overlays and two Packages-Xfce files, One minimal the other full list.
I don't know how much these profiles differ from each other, but it would be easy to add a "minimal" switch, so buildiso decides acording to switch which desktop sub profile it loads.

**Comment by philmmanjaro** [comment]
Also a good idea. Since you're working on a editable package list it should be possible like this. To find out what the exact difference currently is with these profiles just use diff ```diff -Npur <profile> <minimal profile> > diff.txt```

**Comment by udeved** [comment]
Yes, a diff can do that, however, even with diff in a bit like searching the needle in the hay stack.

One downside with integrated minimal profiles, we need a better mechanism to set eg multilib, minimal, or even initsys values, ie to get these vars out of profile.conf.
They would probably better fit in manjaro-tools.conf, since these are effectively user settings, assuming one builds for example openrc profile, while another team member builds same profile with systemd.

**Comment by excalibur1234** [comment]
i have read this thread 2 days ago, but i did not comment. i just wrote a lengthy post and deleted it again. now, i have an opinion:

i do not like the idea of 2 ISO profiles in one. there would be 2 Packages-Xfce files and 2 overlay folders. but then, you also need 2 profile.conf files.
if the minimal version wants to use the CLI installer exclusively, it needs a second Packages-livecd file.
in the end, it would be cluttered ISO profile folder with 2 different versions of most folders and files.
also, we would need another flag/switch/argument in buildiso.

it is much easier to create a new and separate ISO profile. if i do not want to maintain all files in it manually, i can use links to the "shared" profile or any other profile. this is much easier. 
in short: i do not see any benefit.

**Comment by udeved** [comment]
We could have a minimal switch in profile.conf.

The question was more if we can have a shared desktop-overlay wthin one profile, The rest would be done be using a possible >minimal tag in Package list.

**Comment by excalibur1234** [comment]
one example: net-minimal.
net-minimal edition does not use proprietary gpu drivers and pxe (compared to net edition). this means net-minimal would need a custom profile.conf file.

i think it is alright to have a >minimal flag/tag/switch with an option in profile.conf. then,  we could use the >minimal tag/flag/switch in Packages-Xfce and offer 2 versions. 
but this would only be a "minimal light version", because it only offers different packages (but the profile.conf files would be identical). the next step would be to create an argument for buildiso to specify the path of profile.conf. this would make it possible to have 2 different profile.conf files (one for regular and the other for minimal version). the next step would be to have 2 different overlay-xfce folders. the next step would be to have 2 different Packages-Livecd and pacman-default.conf files.

how far do you want to go?

**Comment by udeved** [comment]
~~~
# use plymouth; initcpio hook
# plymouth_boot="true"

# use pxe boot; initcpio hook
# pxe_boot="true"
~~~

We only need to set mkinitcpio vars in devel.

Its just that we might want to move some vars in manjaro-tools.conf or in new file outside version control.

**Comment by excalibur1234** [comment]
i know these options in the devel branch and i like them where they are. i also think that a >minimal tag/switch/flag option in profile.conf is fine. the >minimal switch/flag/tag could be useful in Packages-Xfce or other Packages-... files.

but i think that everything more (a new argument for buildiso or having multiple versions of files or folders in the ISO profile) is bad. as i have already mentioned: a separate profile with symlinks to another profile are easier, more intuitive, and less clutter.
